src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -776,10 +794,48 @@ public enum ClusterDatabaseChangeType
             PendingClusterTransactions
         }
 
-        public void UnloadDirectly(StringSegment databaseName, [CallerMemberName] string caller = null)
+        public void UnloadDirectly(StringSegment databaseName, DateTime? wakeup = null, [CallerMemberName] string caller = null)
         {
+            if (ShouldContinueDispose(databaseName.Value, wakeup, out var dueTime) == false)
+                return;
+
             LastRecentlyUsed.TryRemove(databaseName, out _);
-            DatabasesCache.RemoveLockAndReturn(databaseName, CompleteDatabaseUnloading, out var _, caller).Dispose();
+            DatabasesCache.RemoveLockAndReturn(databaseName, CompleteDatabaseUnloading, out _, caller).Dispose();
+            ScheduleDatabaseWakeup(databaseName.Value, dueTime);
+        }
+
+        private void ScheduleDatabaseWakeup(string name, long milliSeconds)
+        {
+            if (milliSeconds == 0)
+                return;
+
+            _wakeupTimers.TryAdd(name, new Timer(_ =>
+            {
+                if (_disposing.IsWriteLockHeld)
+                    return;
+
+                TryGetOrCreateResourceStore(name);
+            }, null, milliSeconds, Timeout.Infinite));
+        }
+
+        private bool ShouldContinueDispose(string name, DateTime? wakeup, out int dueTime)
+        {
+            dueTime = 0;
+
+            if (name == null)
+                return true;
+
+            if (_wakeupTimers.TryRemove(name, out var timer))
+            {
+                timer.Dispose();
+            }
+
+            if(wakeup.HasValue == false || wakeup.Value == DateTime.MaxValue)
+                return true;
+
+            // if we have a small value or even a negative one, simply don't dispose the database.
+            dueTime = (int)(wakeup - DateTime.Now).Value.TotalMilliseconds;
+            return dueTime > 1000;
```

Comment:
Cost of loading / starting the db is quite high.
Let's skip this if we have a backup scheduled for the next five _minutes_